src/uct/api/uct.h

```diff
@@ -1303,27 +1398,29 @@ ucs_status_t uct_iface_reject(uct_iface_h iface,
  * @ingroup UCT_RESOURCE
  * @brief Create new endpoint.
  *
- * @param [in]  iface   Interface to create the endpoint on.
+ * Create a UCT endpoint in one form available modes:
+ * -# not connected if no any type of address is present, to connect the
+ *    endpoint to remote one need to call @ref uct_ep_connect_to_ep later,
+ *    requires @ref UCT_IFACE_FLAG_CONNECT_TO_EP
+ * -# connected to remote interface if @ref uct_ep_params_t::dev_addr and
+ *    @ref uct_ep_params_t::iface_addr are set, requires iface flag
+ *    @ref UCT_IFACE_FLAG_CONNECT_TO_IFACE
+ * -# connected to socket address if @ref uct_ep_params_t::sockaddr is set,
+ *    requires iface flag @ref UCT_IFACE_FLAG_CONNECT_TO_SOCKADDR
+ *
+ * @param [in]  params  User defined @ref uct_ep_params_t configurations for the
+ *                      @a ep_p.
  * @param [out] ep_p    Filled with handle to the new endpoint.
- */
-ucs_status_t uct_ep_create(uct_iface_h iface, uct_ep_h *ep_p);
-
-
-/**
- * @ingroup UCT_RESOURCE
- * @brief Create an endpoint which is connected to remote interface.
- *
- * requires @ref UCT_IFACE_FLAG_CONNECT_TO_IFACE capability.
  *
- * @param [in]  iface       Interface to create the endpoint on.
- * @param [in]  dev_addr    Remote device address to connect to.
- * @param [in]  iface_addr  Remote interface address to connect to.
- * @param [out] ep_p        Filled with handle to the new endpoint.
+ * @return UCS_OK       The endpoint is created successfully. This does not
+ *                      guarantee that the endpoint has been connected to
+ *                      the destination defined in @a params; in case of failure,
+ *                      the error will be reported to the interface error
+ *                      handler callback provided to @ref uct_iface_open
+ *                      via @ref uct_iface_params_t.err_handler.
+ * @return error code   In case of an error. (@ref ucs_status_t)
```

Comment:
Suggest the following to be more consistent with existing uct and ucp error code descriptions:
`@return` Error code as defined by `@ref` ucs_status_t